app/controllers/chat_channels_controller.rb

```diff
@@ -76,4 +76,34 @@ def notification_settings
       render_json_error(membership)
     end
   end
+
+  def for_tag
+    params.require(:tag_name)
+
+    tag = Tag.find_by(name: params[:tag_name])
+    raise Discourse::NotFound unless tag
+
+    render_channel_for_chatable(
+      ChatChannel.find_by(chatable: tag)
+    )
+  end
+
+  def for_category
+    params.require(:category_id)
+
+    render_channel_for_chatable(
+      ChatChannel.find_by(chatable_id: params[:category_id], chatable_type: "Category")
+    )
+  end
+
+  private
+
+  def render_channel_for_chatable(channel)
+    if channel
+      guardian.ensure_can_see_chat_channel!(channel)
+      render_serialized(channel, ChatChannelSerializer)
+    else
+      render json: { chat_channel: nil }
```

Comment:
I'm curious why we choose to return `nil` here instead of raising a `Discourse::NotFound` error when `channel` is `nil`.

Reply:
No need to error if a category/tag doesn't have a chat_channel. I'd rather just return nil and handle it properly on the client.

Reply:
I understand but I’ve always found it weird from an API perspective. A successful 200 response yet the payload does not contain what I requested for.

Reply:
I don't agree; a nil chat channel is expected behavior, and I don't want to serve a 404 error on expected behavior